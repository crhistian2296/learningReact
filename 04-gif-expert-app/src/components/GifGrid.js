// import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

function GifGrid({ hero }) {
    // const [images, setImages] = useState([]);

    /*
    const [counter, setCounter] = useState(0)

    //useEffect([func], [dependencies])
    */
    // useEffect(() => {
    //     getGifs(hero).then((imgs) => setImages(imgs));
    // }, [hero]);

    //getGifs();

    const { data } = useFetchGifs(hero);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{hero}</h3>

            {/* usando map para imprimir titulos
            <ol>
                {images.map(({ id, title }) => (
                    <li key={id}>{title}</li>
                ))}
            </ol> */}
            {/* Necesarios para ver que hace useEffect
                <h3>{counter}</h3>
                <button type="submit" onClick={() => setCounter(counter + 1)}>+1</button>
            */}
            <div className="gif-container">
                {data.map((img) => (
                    <GifGridItem key={img.id} {...img} className="grid">
                        {
                            // spread de las props de img (herencia)
                        }
                    </GifGridItem>
                ))}
            </div>
        </>
    );
}

GifGrid.propTypes = {
    hero: PropTypes.string.isRequired,
};

export default GifGrid;
