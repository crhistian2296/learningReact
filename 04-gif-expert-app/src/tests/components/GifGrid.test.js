import { shallow } from "enzyme";
import React from "react";
import GifGrid from "../../components/GifGrid";
// jest.mock("../../hooks/useFetchGifs.js");

describe("Pruebas para GifGrid", () => {
    let hero = "Saitama";
    let wrapper = shallow(<GifGrid hero={hero} />);
    beforeEach(() => {
        wrapper = shallow(<GifGrid hero={hero} />);
    });

    test("Debe mostrar el componente correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    //No funciona correctamente por razones desconocidas
    /* 
    test("Debe de mostrar los componentes imgs al renderizar", () => {
        const images = [
            {
                title: "Imagen imaginaria",
                url: "https/localhost/cualquier/cosa/imagen-imaginaria.jpg",
            },
        ];
        useFetchGifs.mockReturnValue({
            data: images,
            loading: false,
        });

        expect(wrapper).toMatchSnapshot();
    }); 
    */
});
