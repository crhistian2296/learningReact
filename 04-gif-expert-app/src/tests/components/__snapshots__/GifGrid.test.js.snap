// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para GifGrid Debe mostrar el componente correctamente 1`] = `
<div
  className="gif-results"
>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Saitama
  </h3>
  <p
    className="animate__animated animate__flash"
  />
  <div
    className="gif-container"
  />
</div>
`;
